src/NodeParser/MappedTypeNodeParser.ts

```diff
@@ -24,23 +24,91 @@ export class MappedTypeNodeParser implements SubNodeParser {
+        P in K
+type :
+        T[P]
+context.arguments:
+        [DefinitionType, UnionType]
+context.parameters:
+        ["T", "K"]
+     */
     private getProperties(node: ts.MappedTypeNode, context: Context): ObjectProperty[] {
-        const type: any = this.typeChecker.getTypeFromTypeNode(node.typeParameter.constraint!);
 
-        if (type.types) {
-            return type.types.map((t: any) => this.getObjectProperty(t, node, context));
-        } else if (type.intrinsicName !== "never") {
-            return [this.getObjectProperty(type, node, context)];
+        const getParameterProperties = function (typeId: string, namesOnly: boolean = false) {
+
+            // @ts-ignore
+            const t = context.arguments.find((v: any, i: any) => {
+                // @ts-ignore
+                return context.parameters[i] === typeId;
+            });
+            // @ts-ignore
+            if (t.type && t.type.properties) { // pick orig
+                // @ts-ignore
+                return t.type.properties;
+                // @ts-ignore
+            } else if (t.types) {  // pick, to pick
+                // @ts-ignore
+                return t.types.map((a: any) => a.value);
+                // @ts-ignore
+            } else if (t.properties)  { // partial orig // partial to pick
+                // @ts-ignore
+                return (namesOnly) ? t.properties.map((p: any) => p.name) : t.properties;
+            }
+
+        };
+
+        // @ts-ignore
+        if (context.parameters.length > 0) {
+            // @ts-ignore
+            const originalProps = (node.type && node.type.objectType) ?
+                // @ts-ignore
+                getParameterProperties(node.type.objectType.typeName.text) : [];
+
+            // @ts-ignore
+            const toPick = (node.typeParameter && node.typeParameter.constraint &&
+                    // @ts-ignore
+                    node.typeParameter.constraint.typeName) ?
+                // @ts-ignore
+                getParameterProperties(node.typeParameter.constraint.typeName.text, true) :
+                // @ts-ignore
+                (node.typeParameter && node.typeParameter.constraint &&
+                // @ts-ignore
+                 node.typeParameter.constraint.type && node.typeParameter.constraint.type.typeName) ?
+                // @ts-ignore
+                getParameterProperties(node.typeParameter.constraint.type.typeName.text, true) :
+                [];
+
+            return originalProps.filter((p: any) => {
+                return toPick.includes(p.name);
+            }).map((p: any) => {
+                p.required = !node.questionToken; // this is for partial
+                return p;
+            });
+
         } else {
-            return [];
+            const type: any = this.typeChecker.getTypeFromTypeNode((<any>node.typeParameter.constraint));
+            if (type.types) {
+                return type.types.reduce((result: ObjectProperty[], t: any) => {
```

Comment:
replace return with map?

Reply:
this second one either needs to be a reduce or a flatMap and I wasn't sure if you wanted to include lodash in the project

Reply:
Why does it need to be a flatMap?

Reply:
because we currently have an array within an array, but we want to end up with just a flat array.

Reply:
I was just reading some of the other code in this project, eg ObjectTypeFormatter - didn't realise you can use destructuring as a flatmap. That's pretty neat.